src/LockstakeEngine.sol

```diff
@@ -218,57 +225,49 @@ contract LockstakeEngine is Multicall {
 
     // --- getters ---
 
-    // NOTE: this function will succeed returning the address even if the urn for the specified index hasn't been created yet
-    function getUrn(address owner, uint256 index) external view returns (address urn) {
-        uint256 salt = uint256(keccak256(abi.encode(owner, index)));
-        bytes32 codeHash = keccak256(abi.encodePacked(_initCode()));
-        urn = address(uint160(uint256(
-            keccak256(
-                abi.encodePacked(bytes1(0xff), address(this), salt, codeHash)
-            )
-        )));
-    }
-
-    function isUrnAuth(address urn, address usr) external view returns (bool ok) {
-        ok = _urnAuth(urn, usr);
+    function isUrnAuth(address owner, uint256 index, address usr) external view returns (bool ok) {
+        ok = _urnAuth(owner, _getUrn(owner, index), usr);
     }
 
     // --- urn management functions ---
 
     function open(uint256 index) external returns (address urn) {
```

Comment:
Don't forget that we support multicall.

So for (1) - The user might not remember how many urns he has and passing the index can avoid mistakes such as opening urn number 3 but depositing to urn number 2.

For (2) - Also here I see cases for confusion since someone might increment the count for the user without him being aware, which may end up locking in the wrong urn. And if the index is still used for validation it can become a DOS vector on `open`s. In general I think this is also the type of change we'd prefer not to do at this point after the Sherlock audit.